List all books on browsing to "/"

qptr opened the pull request.

qptr committed:
Minor changes

qptr committed:
Create jdbc datasource with proper config
Create separate config file for test

qptr committed:
Add api to create and fetch books

qptr committed:
Add handler tests and fixtures

qptr committed:
Fix failing handler test

qptr committed:
Set up migrations with migratus

qptr committed:
Refactor test config management

qptr committed:
Minor refactoring

qptr committed:
Add postgres service,run migrations on github runner

qptr committed:
Run script on staging to create db

qptr committed:
Install postgresql 12 on staging

qptr committed:
Restructure clj directory

qptr committed:
Put staging config into secrets
Pass staging postgres user, password and db as environment variables to script

qptr committed:
Remove create-books handler and tests

qptr committed:
Fix config file path in playbook

qptr committed:
Remove extra quotes around env vars in script

qptr committed:
Fix typo in deploy.yml

qptr committed:
Add domain layer testing
Deploy only on push to master

qptr committed:
Install, set up re-frame 10x

qptr committed:
Dispatch event to manually add books to app-db

qptr committed:
Use re-frame/http-fx to make get rq to fetch books

jysandy commented on test/clj/athenaeum/domain/book_test.clj:
```diff
@@ -0,0 +1,30 @@
+          res (jdbc/with-transaction [tx @db/datasource]
+                (book/fetch-all tx))]
+      (is (= 3 (count res)))
+      (is (= "book1" (:books/title (first res))))
```
We should assert on the entire return value here.

qptr committed:
Add macro to wrap options around transaction object
Add camel-snake-kebab library
Add clear-tables test utility
Edit clj-kondo config for unresolved symbol tx

qptr committed:
Misc refactoring
Add dev util function to run migrations
Extract id-value mapping to private function
Sort books before adding to db

qptr committed:
Dispatch fetch books once on component mount

qptr committed:
Write event handler test with api call stubbing

qptr committed:
Dispatch fetch-books on navigating to page

qptr committed:
Restructure cljs dir, add events and pages dirs

qptr committed:
Add bootstrap table for books

qptr committed:
Add navbar
Move navbar view to separate namespace
Change bootstrap version to 4

qptr committed:
Remove test link from navbar

qptr commented:
Looks like this with a dummy link
<img width="1791" alt="Screen Shot 2021-01-04 at 16 11 21" src="https://user-images.githubusercontent.com/71076330/103527151-ee553a80-4ea7-11eb-91ac-47c0c6cf1369.png">

and like this without
<img width="1791" alt="Screen Shot 2021-01-04 at 16 12 47" src="https://user-images.githubusercontent.com/71076330/103527101-dbdb0100-4ea7-11eb-8cf4-d3c8b0b66b83.png">

qptr committed:
Change year type to smallint

qptr committed:
Minor changes

jysandy commented on src/cljs/athenaeum_web/routes.cljs:
```diff
@@ -5,15 +5,22 @@
         [true :not-found]]])
 
+(def page-list
+  [{:handler :home-page :page-name "Home"}])
```
Let's move this to be alongside the navbar.

jysandy commented on test/clj/athenaeum/fixtures.clj:
```diff
@@ -0,0 +1,24 @@
+
+(defn clear-tables
+  [f]
+  (jdbc/execute! @db/datasource ["TRUNCATE TABLE books"])
```
We can use the `clear-tables` function from test-utils here.

jysandy reviewed (changes requested):
LGTM aside from the suggested changes.

qptr committed:
Changes according to pr comments
